shared/subprocess/manager.go

```diff
@@ -4,19 +4,45 @@
 package subprocess
 
 import (
+	"io"
 	"io/ioutil"
+	"os"
 
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
 )
 
 // NewProcess is a constructor for a process object. Represents a process with argument config. Returns an address to process
 func NewProcess(name string, args []string, stdoutPath string, stderrPath string) (*Process, error) {
+	var stdout, stderr io.WriteCloser
+	var err error
+	// Setup output capture.
+	if stdoutPath != "" {
+		stdout, err = os.Create(stdoutPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to open stdout file")
+		}
+	}
+	if stderrPath == stdoutPath {
+		stderr = stdout
+	} else if stderrPath != "" {
+		stderr, err = os.Create(stderrPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to open stderr file")
+		}
+	}
+
+	return NewProcessWithFds(name, args, nil, stdout, stderr)
+}
+
+// NewProcessWithFds is a constructor for a process object. Represents a process with argument config. Returns an address to process
+func NewProcessWithFds(name string, args []string, stdin io.ReadCloser, stdout io.WriteCloser, stderr io.WriteCloser) (*Process, error) {
```

Comment:
Lets leave stdin out for now as it was not supported before AFAIK.

Reply:
stdin argument is used when `NewProcessWithFds` function is called by `CompressedTarReader` function. So we can't remove it.

Reply:
Makes sense thanks.